test/internet/test-dns-idna2008.js

```diff
@@ -32,13 +32,13 @@ dns.lookup(fixture.hostname, mustCall((err, address) => {
   assert.strictEqual(address, fixture.expectedAddress);
 }));
 
-dns.promises.lookup(fixture.hostname).then(mustCall(({ address }) => {
+dns.promises.lookup(fixture.hostname).then(({ address }) => {
   assert.strictEqual(address, fixture.expectedAddress);
-}).catch((err) => {
+}, (err) => {
   if (err && err.errno === 'ESERVFAIL') {
     assert.ok(err.message.includes('queryA ESERVFAIL straße.de'));
   }
```

Comment:
This error handler causes the test to succeed on any other error. That's not what we want, is it?

Reply:
^^^^ Non-blocking question, not an objection. If I'm right and we don't want that, I'm OK with that being fixed subsequently.

Reply:
Fixed (tried to).
I'll try to create some negative controls.